src/lib/game/world/map.js

```diff
@@ -130,12 +130,17 @@ class WorldMap extends THREE.Group {
         entry.position.y
       );
 
+      // Provided as (Z, X, -Y)
       m2.rotation.set(
+        entry.rotation.z * Math.PI / 180,
         entry.rotation.x * Math.PI / 180,
-        -entry.rotation.z * Math.PI / 180,
-        entry.rotation.y * Math.PI / 180
+        -entry.rotation.y * Math.PI / 180
       );
 
+      // Adjust M2 rotation to match Wowser's axes.
+      const quat = m2.quaternion;
+      quat.set(quat.x, quat.y, quat.z, -quat.w);
+
```

Comment:
Nice! :+1: 

Only thing I'm wondering about is whether it would be possible to calculate the quaternion only? Or do we need to go through Euler first?

THREE.js automatically converts that Euler rotation into a quaternion, here: https://github.com/mrdoob/three.js/blob/master/src/core/Object3D.js#L29-L41